### ISSUE TITLE
[uk] in entries with all kanji forms as search-only

### ISSUE THREAD

**Issue by parfait8566** [JMdictProject/JMdictIssues#152]
I don't see the value of tagging [uk] if the entry has all kanji forms hidden. To me it seems to make sense only for entries that have visible kanji forms through notes.

Also, it'd be nice to have the "[Github Discussions](https://docs.github.com/discussions)" feature enabled to ask questions or have clarifications without necessarily having to create an issue (unless there's already a Discord server or something).

**Comment by JMdictProject** [comment]
I agree that having a [uk] tag is redundant in an information sense when all the kanji forms are hidden, however many apps and sites use the tag to determine the handling and display of the entry. Removing it could cause problems down the line.

**Comment by JMdictProject** [comment]
I have enabled Discussions. Let's see how it goes.

**Comment by parfait8566** [comment]
> I agree that having a [uk] tag is redundant in an information sense when all the kanji forms are hidden, however many apps and sites use the tag to determine the handling and display of the entry. Removing it could cause problems down the line.

What apps and sites? Seems weird such an issue would happen at all if kanji forms are hidden.

> I have enabled Discussions. Let's see how it goes.

Thanks.

**Comment by JMdictProject** [comment]
> Seems weird such an issue would happen at all if kanji forms are hidden.

Well, the "uk" tag has been used for decades, and "sK" only arrived recently, so it had never been an issue before.

We have about 9k entries with "uk" tags, and 72 of them have all the kanji forms tagged as "sK". Not exactly a major problem. I see there are also two entries with all the kanji forms tagged as "sK" but no "uk".

In the case of the WWWJDIC server, I use the "uk" tag to trigger a reformatting; putting the kana forms first, and (unhidden) kanji forms later in parentheses. As it happens, having all the kanji forms hidden has the same outcome.

**Comment by parfait8566** [comment]
Definitely not a major problem, but it is annoying to put [uk] tags just because of search-only kanji forms (especially if there are multiple senses).

> In the case of the WWWJDIC server, I use the "uk" tag to trigger a reformatting; putting the kana forms first, and (unhidden) kanji forms later in parentheses. As it happens, having all the kanji forms hidden has the same outcome.

If the search-only tags are implemented correctly I don't think other sites or apps would have any problem either.

**Comment by JMdictProject** [comment]
Looking back at this one, I guess we can make it so that for entries with [sK] on all the kanji forms, there's no need for [uk] on the readings. Apps can deal with it.

**Comment by stephenmk** [comment]
By my count, there are about 100 entries with [sK] on all the kanji forms and [uk] tags on the senses. If we want to commit to this policy of omitting those [uk] tags, I can go through the list and update them.

Here's the list:

<details>
  <summary>Entry Sequence Numbers</summary>

1000730
1000940
1004120
1004390
1006710
1006980
1007500
1007570
1008790
1010320
1011300
1013020
1030360
1039140
1073270
1081450
1097340
1105280
1108610
1109100
1126030
1374440
1406050
1469800
1565440
1571050
1593920
1612370
1628520
1631370
1632710
1982120
1982300
2003430
2003930
2018320
2025400
2026790
2027080
2034840
2068770
2079190
2085700
2097940
2098080
2119330
2129760
2135530
2137650
2181760
2215120
2219720
2250230
2252690
2337310
2394660
2394710
2436500
2464210
2483630
2554970
2587950
2587960
2592740
2647900
2719290
2755350
2756140
2762760
2763520
2771400
2821200
2823760
2823770
2825840
2826618
2826632
2826811
2829960
2835286
2835552
2837003
2838313
2839040
2848309
2849575
2851570
2857749
2859084
2859097
2859182
2860306
2860695
2861064
2861104
2862048
2863693
2864494
2865116
2865251
2866216
2867141
2867269
</details>

**Comment by JMdictProject** [comment]
I can remove the [uk] tags using a bulk-update script. See entry 1000730 where I have tested this.
If you used a script to generate that list of entry numbers, can you include which senses need treatment? The [uk] tag is on each sense. Entry 1000730 has 7 senses.

**Comment by stephenmk** [comment]
Sure. I'll wait for the recent changes to show up in the XML distribution and then make a CSV file with the entry IDs and sense numbers.

**Comment by JMdictProject** [comment]
FWIW, the format of the updater's command file is:

seq 1000730
del misc[1] uk
del misc[2] uk
del misc[3] uk
del misc[4] uk
del misc[5] uk
del misc[6] uk
del misc[7] uk

**Comment by stephenmk** [comment]
OK, this script should do it. I spot-checked a few and didn't see any problems.

https://gist.github.com/stephenmk/d5d7754cf08e034000e789f44b4ff21c

**Comment by JMdictProject** [comment]
Thanks!  That worked fine. 98 entries updated.
